zipline/utils/factory.py

```diff
@@ -150,27 +152,27 @@ def create_split(sid, ratio, date):
     })
 
 
-def create_txn(sid, price, amount, datetime):
-    txn = Event({
-        'sid': sid,
-        'amount': amount,
-        'dt': datetime,
-        'price': price,
-        'type': DATASOURCE_TYPE.TRANSACTION,
-        'source_id': 'MockTransactionSource'
-    })
-    return txn
+@expect_types(asset=Asset)
+def create_txn(asset, price, amount, datetime, order_id):
```

Comment:
FWIW, if you felt like being fancy, you could just write this as `create_txn = expect_types(asset=Asset)(Transaction)`.